sw/device/examples/aes_serial/aes_serial.c

```diff
@@ -0,0 +1,271 @@
+ * @param plain_text     plain text to be encrypted.
+ * @param plain_text_len plain text length.
+ *
+ * @return SIMPLE_SERIAL_OK on success, SIMPLE_SERIAL_ERROR otherwise.
+ */
+static uint8_t simple_serial_trigger_encryption(const aes_cfg_t *aes_cfg,
+                                                const char *plain_text,
+                                                size_t plain_text_len) {
+  if (plain_text_len > MAX_LENGTH_BIN) {
+    return SIMPLE_SERIAL_ERROR;
+  }
+
+  char cipher_text[MAX_LENGTH_BIN];
+  size_t num_blocks = get_num_blocks(plain_text_len);
+
+  if (dif_gpio_all_write(&gpio, GPIO_SET_TRIGGER_HIGH) != kDifGpioOk) {
+    return SIMPLE_SERIAL_ERROR;
+  }
+
+  aes_init(*aes_cfg);
+  aes_process_blocks((void *)cipher_text, (const void *)plain_text, num_blocks);
+
+  if (dif_gpio_all_write(&gpio, GPIO_SET_TRIGGER_LOW) == kDifGpioOk) {
+    return SIMPLE_SERIAL_ERROR;
+  }
+
+  print_cmd_response('r', cipher_text, plain_text_len);
+  return SIMPLE_SERIAL_OK;
+}
+
+/**
+ * Handle simple seial command.
+ *
+ * @param aes_cfg AES configuration object.
+ * @param cmd     input command buffer.
+ * @param cmd_len number of characters set in input buffer.
+ */
+static void simple_serial_handle_command(const aes_cfg_t *aes_cfg,
+                                         const char *cmd, size_t cmd_len) {
+  char data[MAX_LENGTH_BIN] = {0};
+
+  // Data length is half the size of the hex encoded string.
+  const size_t data_len = cmd_len >> 1;
+  if (data_len >= MAX_LENGTH_BIN) {
+    return;
+  }
+
+  // The simple serial protocol does not expect return status codes for invalid
+  // data, thus it is ok to return early.
+  if (a2b_hex(&cmd[1], data, data_len)) {
+    return;
+  }
+
+  uint8_t result = SIMPLE_SERIAL_OK;
+  switch (cmd[0]) {
+    case 'k':
+      result = simple_serial_set_key(aes_cfg, data, data_len);
+      break;
+    case 'p':
+      result = simple_serial_trigger_encryption(aes_cfg, data, data_len);
+      break;
+    case 'v':
+      result = SIMPLE_PROTOCOL_VERSION;
+      break;
+    default:;
+  }
+  print_cmd_response('z', (const char *)&result, /*data_len=*/1);
+}
+
+int main(int argc, char **argv) {
+  uart_init(kUartBaudrate);
+  base_set_stdout(uart_stdout);
+  pinmux_init();
+
+  dif_gpio_config_t gpio_config = {.base_addr =
+                                       mmio_region_from_addr(0x40010000)};
+
+  // Avoid logging infe
+  if (dif_gpio_init(&gpio_config, &gpio) != kDifGpioOk) {
+    uart_send_str("Fatal error: Unable to initialize GPIO DIF.\r\n");
+    abort();
+  }
+
+  // Enable GPIO 9 and 15 as output.
+  if (dif_gpio_output_mode_all_set(&gpio, 0x08200) != kDifGpioOk) {
+    uart_send_str("Fatal error: Unable to configure GPIO output pins.\r\n");
+    abort();
+  }
+
+  aes_cfg_t aes_cfg;
+  aes_cfg.key_len = kAes128;
+  aes_cfg.operation = kAesEnc;
+  aes_cfg.mode = kAesEcb;
+  aes_cfg.manual_operation = false;
+
+  char text[MAX_LENGTH_TEXT] = {0};
```

Comment:
Again, does this need to be initialized? It looks like you're always writing to this buffer before reading from it.